### PR TITLE
sleve coordimate bugfixes

### PR DESCRIPTION
**Pull request by bertjebertjek**
TYPE: choose one of [bug fix]

KEYWORDS: Sleve coordinate, gamma factor, namelist

SOURCE: Bert Kruyt

DESCRIPTION OF CHANGES: two small bugfixes:
 1. Gamma factor was calculated incorrectly. This did not have an effect on results, as it was only a message on init.
 2. Sleve coordinates were still active in case space_varying_dz=False but SLEVE=True , now fixed.

TESTS CONDUCTED: simple checks, since simple bugs

### Checklist
Merging the PR depends on following checklist being completed. Add `X` between
  each of the square brackets if they are completed in the PR itself. If a
  bullet is not relevant to you, please comment on why below the bullet.

 - [ ] Closes issue #xxxx (An issue must exist or be created to be closed. The
   issue describes and documents the problem and general solution, the PR
   describes the technical details of the solution.)
 - [ ] Tests added (unit tests and/or regression/integration tests)
 - [x] Backwards compatible
 - [ ] Requires new files? If so, how to generate them.
 - [ ] Fully documented

### CODE REVIEW

File: src/objects/domain_obj.f90 [hunk 1]
```diff
@@ -1016,20 +1016,26 @@ subroutine setup_sleve(this, options)
             ! ( Although an argument could be made to calculate this on the offset (u/v) grid b/c that is most
             !   relevant for advection? In reality this is probably a sufficient approximation, as long as we
             !   aren't pushing the gamma factor too close to zero )
-            allocate(gamma_n(this%kds : this%kde+1))
+            ! allocate(gamma_n(this%kds : this%kde+1))
+            allocate(gamma_n(this%kms : max_level))  ! max_level +1 maybe??
+            ! if (options%parameters%debug .and. this_image()==1) write(*,*) " gamma_n: ", gamma_n
```
Comment (gutmann) [inline src/objects/domain_obj.f90:1021]: can you clean up the commented out code?

File: src/objects/domain_obj.f90 [hunk 2]
```diff
@@ -1016,20 +1016,26 @@ subroutine setup_sleve(this, options)
             ! ( Although an argument could be made to calculate this on the offset (u/v) grid b/c that is most
             !   relevant for advection? In reality this is probably a sufficient approximation, as long as we
             !   aren't pushing the gamma factor too close to zero )
-            allocate(gamma_n(this%kds : this%kde+1))
+            ! allocate(gamma_n(this%kds : this%kde+1))
+            allocate(gamma_n(this%kms : max_level))  ! max_level +1 maybe??
```
Comment (gutmann) [inline src/objects/domain_obj.f90:1020]: should it be max_level or kde?

File: src/objects/domain_obj.f90 [hunk 3]
```diff
@@ -1016,20 +1016,26 @@ subroutine setup_sleve(this, options)
...
             gamma_n(i) =  1                                                     &
                 - MAXVAL(h1) * n/(s1**n)                                        &
                 * COSH((smooth_height/s1)**n) / SINH((smooth_height/s1)**n)     &
                 - MAXVAL(h2) * n/(s2**n)                                        &
                 * COSH((smooth_height/s2)**n) / SINH((smooth_height/s2)**n)
+            if (options%parameters%debug .and. this_image()==1) write(*,*) " k, gamma: ", i, gamma_n(i)
 
-            do i = this%grid%kds, this%grid%kde
+            ! do i = this%grid%kds, this%grid%kde 
+            do i = this%grid%kms, max_level-1 ! account for flat z < 0, otherwise gamma can blow up if flat z < 0.
```
Comment (gutmann) [inline src/objects/domain_obj.f90:1032]: does gamma_n get used anywhere or is it purely diagnostic?

File: src/objects/domain_obj.f90 [hunk 4]
```diff
@@ -1124,14 +1130,15 @@ subroutine setup_sleve(this, options)
 
                     endif
 
+                    warn_threshold_dz=0.75
```
Comment (gutmann) [inline src/objects/domain_obj.f90:1133]: this is fine for now, ultimately someone might want to run a simulation with a surface layer 0.75m thick, though it should probably have a small dx for that case too.  Maybe this should search for `dz/dx < threshold` in the future?

File: src/physics/linear_winds.f90 [hunk 5]
```diff
@@ -326,6 +335,8 @@ subroutine linear_perturbation_varyingz(U, V, Nsq, z_bottom, z_top, minimum_step
 
         do while (current_z < end_z)
 
+            ! if (this_image()==1) write(*,*) "      current_z, end_z, step_size:", current_z, end_z, step_size
```
Comment (gutmann) [inline src/physics/linear_winds.f90:338]: cleanup?

File: src/physics/linear_winds.f90 [hunk 6]
```diff
@@ -739,8 +750,9 @@ subroutine initialize_spatial_winds(domain,options,reverse)
 
                         ! call update_irregular_grid(u,v, nsq_values(j), z, domain, minimum_layer_size)
                         call linear_perturbation(u, v, exp(nsq_values(j)),                                                                      &
-                                                 domain%global_z_interface(:,z,:) - domain%global_terrain,                                      &
-                                                 domain%global_z_interface(:,z,:) - domain%global_terrain + domain%global_dz_interface(:,z,:),  &
+                                                 domain%global_z_interface(:,z,:) - domain%global_terrain,                                      &   ! z_bottom
+                                                 domain%global_z_interface(:,z,:) - domain%global_terrain + domain%global_dz_interface(:,z,:),  &   ! z_top
+                                                !  domain%global_z_interface(:,z+1,:) - domain%global_terrain ,  &   ! z_top : Maybe this is a better definition?? should be the same...
```
Comment (gutmann) [inline src/physics/linear_winds.f90:755]: the problem may be that z_interface(z+1) might not exist?

**Review by gutmann — COMMENTED** [review]
minor comments

File: src/objects/domain_obj.f90 (hunk 3 above)
Comment (bertjebertjek) [reply src/objects/domain_obj.f90:1032]: purely diagnostic

File: src/objects/domain_obj.f90 (hunk 2 above)
Comment (bertjebertjek) [reply src/objects/domain_obj.f90:1020]: max level, if there are flat layers above max level these are irellevant to the gamma calculation

File: src/objects/domain_obj.f90 (hunk 1 above)
Comment (bertjebertjek) [reply src/objects/domain_obj.f90:1021]: done

File: src/objects/domain_obj.f90 (hunk 4 above)
Comment (bertjebertjek) [reply src/objects/domain_obj.f90:1133]: agreed

File: src/physics/linear_winds.f90 (hunk 6 above)
Comment (bertjebertjek) [reply src/physics/linear_winds.f90:755]: yes, one would have to adjust the looping index, but i've removed the comment.

**Pull request merged**